### PR TITLE
feat(pipeline): display more readable errors

### PR DESCRIPTION
**Pull request by alice-sevin**
Display errors directly in step.
( errors are going to be filtered from Tucana side, we still return step errors for now ).
Tucana will also return an error in index case (another PR to come, this one need to be merged first)

Before:
before

After:
after

Need to review: https://github.com/ToucanToco/weaverbird/pull/864 first

### CODE REVIEW

**Comment by alice-sevin** [comment]
[ETQC, je différencie visuellement la step sur laquelle survient une erreur sur la vue pipeline](https://trello.com/c/X0i7lJOL/1503-etqc-je-diff%C3%A9rencie-visuellement-la-step-sur-laquelle-survient-une-erreur-sur-la-vue-pipeline)

File: tests/unit/store.spec.ts [hunk 1]
```diff
@@ -336,6 +336,25 @@ describe('getter tests', () => {
...
+      const state = buildState({
+        backendMessages: [{ type: 'error', message: 'lalalolilol', index: 3 }],
+      });
+      expect(getters.stepErrors(state, {}, {}, {})(0)).toBeUndefined();
+    });
+    it('should return undefined if pipeline has no errors', function() {
+      const state = buildState({});
+      expect(getters.stepErrors(state, {}, {}, {})(0)).toBeUndefined();
+    });
+  });
```
Comment (davinov) [inline tests/unit/store.spec.ts:357]: :100:

File: tests/unit/step.spec.ts [hunk 2]
```diff
@@ -239,4 +239,46 @@ describe('Step.vue', () => {
...
+          ],
+        }),
+      );
+      const wrapper = mount(PipelineComponent, { store, localVue });
+      const stepsArray = wrapper.findAll(Step);
+      const replaceStep = stepsArray.at(1);
+      expect(replaceStep.classes()).toContain('query-pipeline-step__container--errors');
+    });
+
+    it('should override with disabled style when disabled', async () => {
```
Comment (davinov) [inline tests/unit/step.spec.ts:285]: Good point! Normally we should never have errors on a disabled step, but in a temporary state when an error is there and we ask for a new preview before the error ; I would never have thought of that case, nice catch :p

File: src/store/actions.ts [hunk 3]
```diff
@@ -26,6 +27,13 @@ function loading(type: 'dataset' | 'uniqueValues') {
   };
 }
 
+// format error to fit BackendError interface props
```
Comment (davinov) [inline src/store/actions.ts:30]: Nice to have extracted that :+1:

**Review by davinov — APPROVED** [review]
So nice! Thanks a lot!

File: tests/unit/store.spec.ts (hunk 1 above)
Comment (alice-sevin) [reply tests/unit/store.spec.ts:357]: Ah mince je viens de voir que j'appelle pas la bonne méthode ligne 344, je corrige ;)

Comment (davinov) [reply tests/unit/store.spec.ts]: Ah bien vu, j'avais pas vu! :eyeglasses:

**Pull request merged**